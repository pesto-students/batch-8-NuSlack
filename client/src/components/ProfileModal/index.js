import React from 'react';
import styled from 'styled-components';

import { Modal } from 'antd';

const StyledModal = styled(Modal)`
  img {
    height: 10em;
    width: 10em;
  }
  .ant-modal-body {
    display: flex;
    flex-direction: row;
  }
  div {
    margin-left: 1em;
  }
`;
const Status = styled.div`
display: inline-block;
background-color: ${props => (props.online === true ? 'green' : 'grey')}
border-radius: 50%;
height: 0.5em;
width: 0.5em;`;
const ProfileModal = props => {
  const { visible, toggleModal, userData } = props;
  return (
    <StyledModal
      visible={visible}
      title="Profile"
      okText="Send Message"
      onCancel={toggleModal}
      onOk={toggleModal}
    >
      <img src={userData.avatar} alt="userImage" />
      <div>
        <h1>
          {userData.title} <Status online />
        </h1>
        <h2>{userData.tagLine}</h2>
      </div>
    </StyledModal>
  );
};

export default ProfileModal;
