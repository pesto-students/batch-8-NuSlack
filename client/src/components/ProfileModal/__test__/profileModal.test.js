import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ProfileModal from '../index';

jest.mock('../index', () => () => <profile-modal />);

describe('<ProfileModal />', () => {
  it('should match the snapshot', () => {
    const tree = shallow(<profile-modal />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
