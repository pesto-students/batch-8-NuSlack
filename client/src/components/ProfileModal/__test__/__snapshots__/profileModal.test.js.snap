// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileModal /> should match the snapshot 1`] = `<profile-modal />`;
